### PR TITLE
golangci-lint and npm-check build steps with various code improvements

### PR DESCRIPTION
**Pull request by jasonMcCullough**
## Goal

* Aims to improve code quality with the inclusion of `golangci-lint` and `npm-check` into our CI build process for the BugSnag CLI.
* Mitigates pre-existing code smells and other lint violations flagged by `golangci-lint`.
* Mitigates old or unused etc dependencies found in `package.json` via `npm-check`

## Design

* A similar approach is used by another team in BugSnag which inspired the incorporation of this into this repo.
* Leveraging `npm-check` for JS dependencies was easy to get up and running and was easy to fix using it's prompted directives.

## Changeset

* No functionality in the code has changed other than how we parse supported platforms which has now been exported into an easy to locate and edit `.yml` file

## Testing

* Pre-existing tests were used to validate that:
  * No functionality was broken in the process of this change.
  * No un-wanted/un-expected functionality changes were made as a result.
* These changes were ran and tested locally by building and running the CLI tool locally along with running the make file locally with `golangci-lint`.

### CODE REVIEW

**Review by joshedney — COMMENTED** [review]
Mainly looking good. Just a few small comments

**Comment by joshedney** [comment]
If you could also add a PR description

File: pkg/upload/dart.go [hunk 1]
```diff
@@ -26,28 +28,28 @@ type DartSymbolOptions struct {
 
 func Dart(paths []string, version string, versionCode string, bundleVersion string, iosAppPath string, endpoint string, timeout int, retries int, overwrite bool, apiKey string, failOnUploadError bool, dryRun bool) error {
 
-	log.Info("Building file list from path")
```
[the remark this answers is not in the document]
Comment (jasonMcCullough) [reply pkg/upload/dart.go]: Appears as though it was a find and replace gone wrong 😅 I'll change it all back as looking back, was an unnecessary change

File: install.js [hunk 2]
```diff
@@ -17,44 +19,8 @@ const removeGitPrefixAndSuffix = (input) => {
     return result;
 };
 
-const supportedPlatforms = [
```
[the remark this answers is not in the document]
Comment (jasonMcCullough) [reply install.js:20]: You're right, this was tested locally but I did not check if `npm publish` and the like were performing as expected. I've now included this into `package.json` so we can package the file with everything else.

File: pkg/upload/dart.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshedney) [reply pkg/upload/dart.go]: I assumed it was :D

File: package.json [hunk 3]
```diff
@@ -18,10 +18,12 @@
...
   "dependencies": {
     "axios": "^1.4.0",
-    "console.table": "^0.10.0"
+    "console.table": "^0.10.0",
+    "js-yaml": "^4.1.0"
   },
   "files": [
-    "install.js"
+    "install.js",
+    "yml"
```
Comment (joshedney) [inline package.json:26]: Does this need to be `supported-platforms.yml` ?

Comment (jasonMcCullough) [reply package.json]: Ah, it should be. I thought including just yml would do all files with that extension. It didn't! But after making this change it works:
```
 > tar -tvf bugsnag-cli-2.0.0.tgz                                                                                                                                     
-rw-r--r--  0 0      0        1051 Oct 26  1985 package/LICENSE
-rw-r--r--  0 0      0        2557 Oct 26  1985 package/install.js
-rw-r--r--  0 0      0         667 Oct 26  1985 package/package.json
-rw-r--r--  0 0      0        5564 Oct 26  1985 package/README.md
-rw-r--r--  0 0      0         695 Oct 26  1985 package/supported-platforms.yml
```

**Review by joshedney — APPROVED** [review]
LGTM!

**Pull request merged**